task/04-arrays-tasks.js

```diff
@@ -469,7 +624,14 @@ function getIntervalArray(start, end) {
  *   [ 1, 1, 2, 2, 3, 3, 4, 4] => [ 1, 2, 3, 4]
  */
 function distinct(arr) {
-   throw new Error('Not implemented');
+   let result = [];
+   arr.map(function (elem) {
+      if (result.indexOf(elem) === -1) {
```

Comment:
you are using the `map` as loop here. That's a hack for the task ) please try with reduce instead